narwhals/_pandas_like/utils.py

```diff
@@ -325,57 +321,46 @@ def vertical_concat(dfs: list[Any], implementation: str) -> Any:
     if not dfs:
         msg = "No dataframes to concatenate"  # pragma: no cover
         raise AssertionError(msg)
+
+    backend = get_backend(implementation)
     cols = set(dfs[0].columns)
     for df in dfs:
         cols_current = set(df.columns)
         if cols_current != cols:
             msg = "unable to vstack, column names don't match"
             raise TypeError(msg)
-    if implementation == "pandas":
-        pd = get_pandas()
+    if implementation is Implementation.PANDAS:
+        pandas_version = parse_version(backend.__version__)
 
-        if parse_version(pd.__version__) < parse_version("3.0.0"):
-            return pd.concat(dfs, axis=0, copy=False)
-        return pd.concat(dfs, axis=0)  # pragma: no cover
-    if implementation == "cudf":  # pragma: no cover
-        cudf = get_cudf()
+        if pandas_version < parse_version("3.0.0"):
+            return backend.concat(dfs, axis=0, copy=False)
+        return backend.concat(dfs, axis=0)  # pragma: no cover
 
-        return cudf.concat(dfs, axis=0)
-    if implementation == "modin":  # pragma: no cover
-        mpd = get_modin()
+    if (
+        implementation is Implementation.CUDF or implementation is Implementation.MODIN
+    ):  # pragma: no cover
+        return backend.concat(dfs, axis=0)
 
-        return mpd.concat(dfs, axis=0)
-    msg = f"Unknown implementation: {implementation}"  # pragma: no cover
-    raise TypeError(msg)  # pragma: no cover
+    return assert_never(implementation)
 
 
 def native_series_from_iterable(
-    data: Iterable[Any], name: str, index: Any, implementation: str
+    data: Iterable[Any], name: str, index: Any, implementation: PANDAS_IMPLEMENTATIONS
 ) -> Any:
     """Return native series."""
-    if implementation == "pandas":
-        pd = get_pandas()
-
-        return pd.Series(data, name=name, index=index, copy=False)
-    if implementation == "cudf":  # pragma: no cover
-        cudf = get_cudf()
-
-        return cudf.Series(data, name=name, index=index)
-    if implementation == "modin":  # pragma: no cover
-        mpd = get_modin()
+    series_implementation = get_series_implementation(implementation)
 
-        return mpd.Series(data, name=name, index=index)
-    msg = f"Unknown implementation: {implementation}"  # pragma: no cover
-    raise TypeError(msg)  # pragma: no cover
+    return series_implementation(data, name=name, index=index)
 
 
-def set_axis(obj: T, index: Any, implementation: str) -> T:
-    if implementation == "pandas" and parse_version(
-        get_pandas().__version__
-    ) >= parse_version("1.5.0"):
+def set_axis(obj: T, index: Any, implementation: PANDAS_IMPLEMENTATIONS) -> T:
+    backend = get_backend(implementation)
+    backend_verison = parse_version(backend.__version__)
+    minimum_version = parse_version("1.5.0")
+    if implementation is Implementation.PANDAS and backend_verison >= minimum_version:
```

Comment:
(similar nitpicking)

```suggestion
    if implementation is Implementation.PANDAS and (
        backend_verison >= parse_version("1.5.0")
    ):
```